src/main/Duke/virtue/VirtueTaskList.java

```diff
@@ -0,0 +1,168 @@
+package virtue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VirtueTaskList {
+    // The task list.
+    private List<VirtueTask> taskList;
+
+    public VirtueTaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    // Gets the number of tasks in the task list.
+    private int numTasks() {
+        return taskList.size();
+    }
+
+    // Gets the i-th task from the task list, where i is the index.
+    private VirtueTask getTask(int index) {
+        return taskList.get(index - 1);
+    }
+
+    // Adds a todo task to the task list.
+    private void addTodo(String description) {
```

Comment:
Method names like addTodo, addDeadline, addEvent, markTaskAsDone, and printOut clearly convey their functionality, which is excellent for code readability.